pkg/gatewayserver/io/basicstationlns/basicstationlns.go

```diff
@@ -183,20 +200,57 @@ func (s *srv) handleTraffic(c echo.Context) error {
 				"firmware", version.Firmware,
 				"model", version.Model,
 			))
-			fp, err := s.server.GetFrequencyPlan(ctx, ids)
+			cfg, err := messages.GetRouterConfig(*fp, version.IsProduction())
 			if err != nil {
-				logger.WithError(err).Warn("Failed to get frequency plan")
+				logger.WithError(err).Warn("Failed to generate router configuration")
+				return err
+			}
+			data, err = json.Marshal(cfg)
+			if err != nil {
+				logger.WithError(err).Warn("Failed to marshal response message")
+				return err
+			}
+			if err := ws.WriteMessage(websocket.TextMessage, data); err != nil {
+				logger.WithError(err).Warn("Failed to send router configuration")
 				return err
 			}
-			// TODO: Send frequency plan, see messages.RouterConfig.
-			_ = fp
 
-		// TODO: Add case for uplink messages.
-		case messages.TypeJoinRequest:
-			// TODO
+		case messages.TypeUpstreamJoinRequest:
+			var jreq messages.JoinRequest
+			if err := json.Unmarshal(data, &jreq); err != nil {
+				logger.WithError(err).Debug("Failed to unmarshal join request message")
+				return err
+			}
+			up, err := jreq.ToUplinkMessage(ids, fp.BandID)
+			if err != nil {
+				logger.WithError(err).Debug("Failed to parse join request message")
+				return err
+			}
+			if err := conn.HandleUp(&up); err != nil {
+				logger.WithError(err).Warn("Failed to handle uplink message")
+			}
+
+		case messages.TypeUpstreamJoinUplinkDataFrame:
+		case messages.TypeUpstreamTxConfirmation:
+		case messages.TypeUpstreamProprietaryDataFrame:
+		case messages.TypeUpstreamRemoteShell:
+		case messages.TypeUpstreamTimeSync:
 
 		default:
+			// Unknown message types are ignored by the server
 			logger.WithField("message_type", typ).Debug("Unknown message type")
 		}
 	}
 }
+
+// sendDiscoveryErrorMessageToStation sends the error messages during the discovery on the WS connection to the station.
+func sendDiscoveryErrorMessageToStation(ws *websocket.Conn, logger log.Interface, msg string) {
```

Comment:
- `writeDiscoverError()` ?
- Pass the context and get the logger from the context
- Simpler is just return the error and only (debug) log causes, if relevant